rfc/rfc-51/rfc-51.md

```diff
@@ -64,71 +65,74 @@ We follow the debezium output format: four columns as shown below
+|-----------------------------------------------------|----------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| hoodie.table.cdc.enabled                            | `false`  | The master switch of the CDC features. If `true`, writers and readers will respect CDC configurations and behave accordingly.                                                                                                                                                                                        |
+| hoodie.table.cdc.supplemental.logging.mode          | `KEY_OP` | A mode to indicate the level of changed data being persisted. At the minimum level, `KEY_OP` indicates changed records' keys and operations to be persisted. `DATA_BEFORE`: persist records' before-images in addition to `KEY_OP`. `DATA_BEFORE_AFTER`: persist records' after-images in addition to `DATA_BEFORE`. |
 
-![](arch.jpg)
+To perform CDC queries, users need to set `hoodie.datasource.query.incremental.format=cdc` and `hoodie.datasource.query.type=incremental`.
 
-Note: Table operations like `Compact`, `Clean`, `Index` do not write/change any data. So we don't need to consider them in CDC scenario.
- 
-### Modifiying code paths
+| key                                        | default        | description                                                                                                                          |
+|--------------------------------------------|----------------|--------------------------------------------------------------------------------------------------------------------------------------|
+| hoodie.datasource.query.type               | `snapshot`     | set to `incremental` for incremental query.                                                                                          |
+| hoodie.datasource.query.incremental.format | `latest_state` | `latest_state` (current incremental query behavior) returns the latest records' values. Set to `cdc` to return the full CDC results. |
+| hoodie.datasource.read.start.timestamp     | -              | requried.                                                                                                                            |
+| hoodie.datasource.read.end.timestamp       | -              | optional.                                                                                                                            |
```

Comment:
`hoodie.datasource.read.start.timestamp`
`hoodie.datasource.read.end.timestamp`

Would these two options be needed by the fs view API or only for reader/writer ? The `start.timestamp` should also have a default value and by default, consumes from the latest commit.

Reply:
Why would ```start.timestamp``` have a default value? This is essentially the consumer marker and this is stored outside of Hudi. ```end.timestamp``` default value should be the latest commit.

Reply:
> Why would start.timestamp have a default value?

A default value as latest commit keeps more align with message queue like Kafka, by default, they consumes from the latest for the consumer.

> end.timestamp default value should be the latest commit.

Agree with this but only for the case that end.timestamp exists which means it is a batch query, remember that we also have streaming query, there is no real end.timestamp in this case.

Reply:
These are not new configs. We're referring to existing configs here

- https://hudi.apache.org/docs/configurations#hoodiedatasourcereadbegininstanttime
- https://hudi.apache.org/docs/configurations#hoodiedatasourcereadendinstanttime (docs should be updated to say optional instead of required)

I'll also update the config keys to using "begin.instanttime" and "end.instanttime"

For whether to have default value for begin instant time or not, I'm also inclined to have it required and keep the user behavior unchanged. It might improve the UX if we set a default like latest time, but it's a separate topic from this RFC.